### PR TITLE
refactor(hostinfo): Make the api consistent across hostinfo checks

### PR DESCRIPTION
**Pull request by kaustavha**
Make these checks return a list of objects, stop storing data like filenames in keys, instead we'll go with having a key called name which stores that data instead. And we'll log warning instead of sending them downstream.

### CODE REVIEW

File: hostinfo/misc.lua [hunk 1]
```diff
@@ -132,5 +135,24 @@ local function asyncSpawn(dataArr, spawnFunc, successFunc, finalCb)
...
+local function logWarn(errTable)
+  local logpath
+  if los.type() ~= 'linux' then
+    -- TODO: properly implement logging in windows
+    logpath = './'
+  else
+    logpath = '/var/log'
+  end
+  logpath = join(logpath, 'rackspace-monitoring-agent-hostinfo.log')
+  logging.init(logging.StdoutFileLogger:new({
```
Comment (rphillips) [inline hostinfo/misc.lua]: shouldn't have to do this

Comment (rphillips) [inline hostinfo/misc.lua]:
```
logging.logf(logging.WARN, "  libuv %s", uv.version_string())
```

Comment (kaustavha) [inline hostinfo/misc.lua]: The file that contains your line there(main.lua) inits the logger with a logpath derived from the -l argument to the cli. 
```
  -- Setup Logging
  logging.init(logging.StdoutFileLogger:new({
    log_level = log_level,
    path = argv.args.l
  }))
```
Trying to run the logger without initing it with a  logpath doesnt work. (missing lib err || missing numeric literal at line 1 column 6).
Trying to run logging.logf without initing the logger results in the following err:
```
Uncaught exception:
[string "bundle:deps/logging/init.lua"]:92: attempt to compare number with nil
```

Comment (rphillips) [inline hostinfo/misc.lua]: [This](https://gist.github.com/rphillips/31d8300e85445d6d9c71) works

Comment (kaustavha) [inline hostinfo/misc.lua]: screen shot 2015-08-04 at 1 49 44 pm
screen shot 2015-08-04 at 1 50 46 pm

Comment (rphillips) [inline hostinfo/misc.lua]: this is expected... the pipe to jq will try and parse the log message, fail and say there is a json error

**Pull request closed without merging**